src/app/api/article/route.ts

```diff
@@ -16,9 +17,16 @@ export const POST = async (request: NextRequest) => {
     return new Response("Unauthorized", { status: 401 });
   }
 
+  const articleId = nanoid(6);
+
+  // we might allow user to set their own slug?
+  const slug = concat(slugify(body.slug || body.title), "-", articleId);
```

Comment:
This can be problematic in the future. Adding article id inside the slug has two disadvantages:
- The old article without slug will not function unless in the migration script you mass update the slug
- If we allow user to change their own slug then append the article id at the end will be troublesome.

Reply:
ahh i didnt think of that, will remove that, then when linking to an article we do this instead `/@${username}/${concat(slug, "-", id)}`?

Reply:
That would be more flexible.